src/main/ngapp/src/app/main/main.component.html

```diff
@@ -2,60 +2,65 @@
-          nbContextMenuTag="my-context-menu"
-        ></nb-action>
-      </nb-actions>
+      <button nbButton size="large" class="pull-right" [nbContextMenu]="menuItems" nbContextMenuTag="my-context-menu">
+          <nb-icon icon="person-outline" status="primary" size="large"></nb-icon>
+      </button>
     </div>
   </nb-layout-header>
 
-  <ng-template #dialog let-data let-ref="dialogRef">
-    <nb-card class="dialog-template">
-      <nb-card-header class="header-container" (click)="ref.close()">
-        Update Language Preference
-        <button nbSuffix nbButton ghost autofocus>
-          <nb-icon icon="close"></nb-icon>
-        </button>
-      </nb-card-header>
-      <nb-card-body>
-        Kindly update your language
-        <nb-select
-          placeholder="Select Showcase"
-          ngModel
-          #language="ngModel"
-          name="language"
-          id="language"
-          required
-        >
-          <nb-option value="en">English</nb-option>
-          <nb-option value="fr">French</nb-option>
-          <nb-option value="sp">Spanish</nb-option>
-          <nb-option value="it">Italian</nb-option>
-          <nb-option value="sw"> Swahili </nb-option>
-          <nb-option value="zh">Chinese</nb-option>
-          <nb-option value="ar">Arabic </nb-option>
-          <nb-option value="de">German </nb-option>
-        </nb-select>
-      </nb-card-body>
-      <nb-card-footer>
-        <button
-          nbButton
-          status="primary"
-          [disabled]="language.invalid"
-          (click)="updateLanguage(language.value)"
-        >
-          Update
-        </button>
-      </nb-card-footer>
-    </nb-card>
-  </ng-template>
+  <nb-layout-column [nbSpinner]="mainLoading" nbSpinnerStatus="primary">
+
+    <nb-alert status="danger" *ngIf="languageUpdateError === true">
+        {{ languageUpdateErrorMessage }}
+    </nb-alert>
+
+    <ng-template #dialog let-data let-ref="dialogRef">
+        <nb-card class="dialog-template">
+        <nb-card-header class="header-container">
+            Update Language Preference
+            <button nbSuffix nbButton ghost autofocus (click)="ref.close(); closeAlert()">
+                <nb-icon icon="close"></nb-icon>
+            </button>
+        </nb-card-header>
+        <nb-card-body>
+            <nb-alert status="danger" closable (close)="closeAlert()" *ngIf="languageUpdateError === true">
+                {{ languageUpdateErrorMessage }}
+            </nb-alert>
+            <nb-alert status="success" closable (close)="closeAlert()" *ngIf="languageUpdateError === false">
+                Success
+            </nb-alert>
+            Kindly update your language
+            <nb-select
+            placeholder="Select Language"
+            ngModel
+            #language="ngModel"
+            name="language"
+            id="language"
+            required
+            >
+            <nb-option value="en">English</nb-option>
```

Comment:
iterate to generate options. Putting lang choices in to an array.

Reply:
implemented